### ISSUE TITLE
Make optional the analyze part in CDB_EstimateRowCount

### ISSUE THREAD

**Issue by ethervoid** [CartoDB/cartodb-postgresql#301]
The CDB_EstimateRowCount function executes an `ANALYZE` in the affected tables of the query when that tables don't have stats. But this operation could come at a great cost if, for example, the table is big so we should add an optional parameter to skip the analyze part where performance is more important and we can deal with tables without stats.

This issue arises because in the tiler we're using this function to gather stats from the layer queries and execute an ANALYZE could penalize the performance

**Comment by rafatower** [comment]
is this required to go out with vector/raster?

**Comment by ethervoid** [comment]
Due to a conversation with @javisantana I think is a nice to have more than a must

**Comment by javisantana** [comment]
I don't think this is critical but there are a few things why we should change this:

- writing in a function that is created to get data seems not fine. 
- calling this function with different users could give different results
- depending on that statistics target it cold take a while and this function was created to be fast (that's why we are estimating and not counting)
